src/main/java/com/contentful/java/cda/CDAHttpException.java

```diff
@@ -37,6 +40,9 @@ public CDAHttpException(Request request, Response response) {
 
   private String readResponseBody(Response response) {
     try {
+      BufferedSource bufferedSource = response.body().source();
+      Timeout timeout = bufferedSource.timeout();
+      timeout.deadline(1, TimeUnit.SECONDS);
```

Comment:
This is only used when an error body is parsed. Our error bodies should not be that big that a timeout of a second gets hit. Additionally I think, if the connection is that bad, crashing at the error body parsing level will also indicate the same problem.